harness/determined/common/api/request.py

```diff
@@ -356,3 +363,37 @@ def ws(host: str, path: str) -> WebSocket:
     token = authentication.must_cli_auth().get_session_token()
     websocket.add_header("Authorization".encode(), "Bearer {}".format(token).encode())
     return WebSocket(websocket)
+
+
+def parse_warnings(
+    warnings: Optional[Union[Sequence[bindings.v1LaunchWarning], Sequence[int]]]
+) -> Optional[Sequence[bindings.v1LaunchWarning]]:
+
+    # Warnings may be passed in as either an enum value or integer value
+    warning_message_int_to_enum_value = {
+        1: bindings.v1LaunchWarning.LAUNCH_WARNING_CURRENT_SLOTS_EXCEEDED
+    }
+
+    warning_enum_values: list[bindings.v1LaunchWarning] = []
+    if warnings:
+        for warning in warnings:
+            if isinstance(warning, int):
+                warning_value: int = warning
+                warning_enum_values.append(warning_message_int_to_enum_value[warning_value])
+            elif isinstance(warning, str):
+                warning_enum_value: bindings.v1LaunchWarning = bindings.v1LaunchWarning(warning)
+                warning_enum_values.append(warning_enum_value)
+
+    return warning_enum_values
+
+
+def handle_warnings(warnings: Optional[Sequence[bindings.v1LaunchWarning]]) -> None:
+    warning_message_map = {
+        bindings.v1LaunchWarning.LAUNCH_WARNING_CURRENT_SLOTS_EXCEEDED: (
+            "Warning: The requested job requires more slots than currently available. "
+            "You may need to increase cluster resources in order for the job to run."
+        )
+    }
```

Comment:
Can you make this map a constant in `common/api/_util.py`, then expose it in the `determined.common.api` namespace by importing it into the `common/api/__init__.py` file?  (see the pattern in that file).  This constant belongs there because it is useful to both CLI and non-CLI code.

Then can you separate the printing into a CLI-specific function.  Put that in `cli/_util.py`, then expose it into the `determined.cli` namespace by importing it in the `cli/__init__.py` file.  This particular function is only useful to CLI functions because `print()` shouldn't be called by library code (so, nothing in `determined.experimental.client`, for instance).  But the CLI is an application, not a library, and printing yellow errors is totally fine there.

Reply:
updated